src/main/kotlin/com/ekino/oss/gradle/plugin/docker/DockerPlugin.kt

```diff
@@ -1,47 +1,51 @@
 package com.ekino.oss.gradle.plugin.docker
 
-import com.avast.gradle.dockercompose.ComposeSettings
+import com.avast.gradle.dockercompose.ComposeExtension
 import org.gradle.api.Plugin
 import org.gradle.api.Project
+import org.gradle.api.Task
 import org.gradle.api.tasks.JavaExec
 import org.gradle.api.tasks.testing.Test
 import java.io.File
 
 class DockerPlugin : Plugin<Project> {
 
   override fun apply(project: Project) {
-    // plugins
-    project.pluginManager.apply("docker-compose")
+    with(project) {
+      if (!File("${projectDir}/docker-compose.yml").exists()) {
+        return
+      }
+
+      // plugins
+      pluginManager.apply("docker-compose")
 
-    if (File("${project.projectDir}/docker-compose.yml").exists()) {
-      val dockerCompose = ComposeSettings(project)
+      val dockerCompose = extensions.getByName("dockerCompose") as ComposeExtension
```

Comment:
Hi, `project.extensions.getByType(ComposeExtension::class.java)` seems better :)

Reply:
Yeah you're right. It's done.